bolt-simple-taproot.md

```diff
@@ -0,0 +1,1141 @@
+
+#### To Remote Outputs
+
+As we inherit the anchor output semantics we want to ensure that the remote
+party can unilaterally sweep their funds after the 1 block CSV delay. In order
+to achieve this property, we'll re-use the `_funding_key` here: its in
+the best interest of the other party to enforce these semantics (mempool
+pinning mitigation), so the remote party will be forced to always take the
+script reveeal path.
+
+The to remote output has the following form:
+
+  * `OP_1 to_remote_output_key`
+  * where:
+    * `to_remote_output_key = combined_funding_key + tagged_hash("TapTweak", combined_funding_key || to_remote_script_root)`
+    * `to_remote_script_root = tapscript_root([to_remote_script])`
+    * `to_remote_script` is the remote script:
+        ```
+        <remotepubkey> OP_CHECKSIG
+        OP_CHECKSEQUENCEVERIFY
+        ```
+
+This output can be swept by the remote party with the following witness:
+```
+<remote_sig> <to_remote_script> <to_remote_control_block>
+```
+
+where `to_remote_control_block` is:
+```
+(output_key_y_parity | 0xc0) || combined_funding_key
+```
+
+#### Anchor Outputs
+
+For simple taproot channels (`option_simple_taproot`) we'll maintain the same
+form as base segwit channels, but instead will utilize `local_delayedpubkey`
+and the `remotepubkey` rather than the multi-sig keys as that's no longer
+revealed due to musig2.
+
+An anchor output has the following form:
+
+  * `OP_1 anchor_output_key`
+  * where:
+    * `anchor_internal_key = remotepubkey/local_delayedpubkey`
+    * `anchor_output_key = anchor_internal_key + tagged_hash("TapTweak", anchor_internal_key || anchor_script_root)`
+    * `anchor_script_root = tapscript_root([anchor_script])`
+    * `anchor_script`:
+        ```
+        OP_16 OP_CHECKSEQUENCEVERIFY
+        ```
+
+This output can be swept by anyone after 16 blocks with the following witness:
+```
+<anchor_script> <anchor_control_block>
+```
+
+where `anchor_control_block` is:
+```
+(output_key_y_parity | 0xc0) || anchor_internal_key
+```
+
+`nSequence` needs to be set to 16.
+
+### HTLC Scripts & Transactions
+
+#### Offered HTLCs
+
+We retain the same second-level HTLC mappings as base channels. The main
+modifications are using the revocation public key as the internal key, and
+splitting the timeout and success paths into individual script leaves.
+
+An offered HTLC has the following form:
+
+  * `OP_1 offered_htlc_key`
+  * where:
+    * `offered_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <local_htlcpubkey> OP_CHECKSIGVERIFY
+        <remote_htlcpubkey> OP_CHECKSIG
+        ```
+    * `htlc_success`:
+        ```
+        OP_SIZE 32 OP_EQUALVERIFY OP_HASH160 <RIPEMD160(payment_hash)> OP_EQUALVERIFY
+        <remote_htlcpubkey> OP_CHECKSIG
+        OP_CHECKSEQUENCEVERIFY
```

Comment:
my 2 cents only but don't think saving 1 WU per onchain success tx is worth it. miniscript does the expected variant:
```
and(hash160(H),and(pk(A),after(1)))
```

```
OP_SIZE <20> OP_EQUALVERIFY OP_HASH160 <h> OP_EQUALVERIFY <A> OP_CHECKSIGVERIFY
1 OP_CHECKLOCKTIMEVERIFY
```

Reply:
was discussed here: https://github.com/Roasbeef/lightning-rfc/pull/1#discussion_r886414660

Reply:
the script now in the bolt does the same thing as your script, the difference is the on-chain footprint (weight units)

Reply:
I understand it functionally completes the same; I strongly suspect 1 WU is not worth the other costs, such as readability, analysis, miniscript compatbility(which is related).

It's a really interesting trick though!

Reply:
Yeah we understand that the behavior is the same, the discussion here now is if we should just have that extra byte which makes things much easier to understand at a glance (this has come up several times in code review so far in https://github.com/lightningnetwork/lnd/pulls).

Reply:
Not sure what decision was made here but I think it would make much more sense to save the on-chain space and be compatible with miniscript